From: Hannes Reinecke <hare@suse.de>
Date: Fri, 8 Mar 2024 08:11:05 +0100
Subject: nvmet-tcp: do not continue for invalid icreq
Patch-mainline: v6.9-rc1
Git-commit: 0889d13b9e1cbef49e802ae09f3b516911ad82a1
References: git-fixes

When the length check for an icreq sqe fails we should not
continue processing but rather return immediately as all
other contents of that sqe cannot be relied on.

Signed-off-by: Hannes Reinecke <hare@suse.de>
Reviewed-by: Christoph Hellwig <hch@lst.de>
Reviewed-by: Sagi Grimberg <sagi@grimberg.me>
Signed-off-by: Keith Busch <kbusch@kernel.org>
Acked-by: Daniel Wagner <dwagner@suse.de>
---
 drivers/nvme/target/tcp.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/nvme/target/tcp.c
+++ b/drivers/nvme/target/tcp.c
@@ -897,6 +897,7 @@ static int nvmet_tcp_handle_icreq(struct
 		pr_err("bad nvme-tcp pdu length (%d)\n",
 			le32_to_cpu(icreq->hdr.plen));
 		nvmet_tcp_fatal_error(queue);
+		return -EPROTO;
 	}
 
 	if (icreq->pfv != NVME_TCP_PFV_1_0) {
